### PR TITLE
\yii\log\FileTarget directory creation is moved to export() method

### PR DESCRIPTION
**Pull request by alexeevdv**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15662

### CODE REVIEW

**Review by SilverFire — APPROVED** [review]

File: framework/log/FileTarget.php [hunk 1]
```diff
@@ -105,6 +101,11 @@ public function init()
      */
     public function export()
     {
+        $logPath = dirname($this->logFile);
+        if (!is_dir($logPath)) {
+            FileHelper::createDirectory($logPath, $this->dirMode, true);
+        }
```
Comment (cebe) [inline framework/log/FileTarget.php]: `export()` can be triggered multiple times in the application flow, its probably better to do it in `init()` but only if `enabled` is true.

Comment (cebe) [reply framework/log/FileTarget.php]: thinking more about it, my suggestion will make it much more complex as the path should still be created after enabling a disabled log target. Looks good to me as is.

**Review by cebe — APPROVED** [review]

**Comment by SilverFire** [comment]
Btw, `is_dir` could be changed to `file_exists` as it works a bit faster

**Comment by alexeevdv** [comment]
`is_dir` check is not needed actually. Since it is performed inside `FileHelper::createDirectory` anyway

**Review by samdark — APPROVED** [review]

**Pull request merged**